src/TurbulenceClosures/turbulence_closure_implementations/smagorinsky_lilly.jl

```diff
@@ -96,15 +96,25 @@ filter width `Δᶠ`, and strain tensor dot product `Σ²`.
 """
 @inline νₑ_deardorff(ς, C, Δᶠ, Σ²) = ς * (C*Δᶠ)^2 * sqrt(2Σ²)
 
-@inline function calc_nonlinear_νᶜᶜᶜ(i, j, k, grid::AbstractGrid{FT}, clo::SmagorinskyLilly, buoyancy, U, C) where FT
+@inline function calc_nonlinear_νᶜᶜᶜ(i, j, k, grid::AbstractGrid{FT}, closure::SmagorinskyLilly, buoyancy, U, C) where FT
     Σ² = ΣᵢⱼΣᵢⱼᶜᶜᶜ(i, j, k, grid, U.u, U.v, U.w)
     N² = max(zero(FT), ℑzᵃᵃᶜ(i, j, k, grid, ∂z_b, buoyancy, C))
-    Δᶠ = Δᶠ_ccc(i, j, k, grid, clo)
-    ς  = stability(N², Σ², clo.Cb) # Use unity Prandtl number.
+    Δᶠ = Δᶠ_ccc(i, j, k, grid, closure)
+    ς  = stability(N², Σ², closure.Cb) # Use unity Prandtl number.
 
-    return νₑ_deardorff(ς, clo.C, Δᶠ, Σ²)
+    return νₑ_deardorff(ς, closure.C, Δᶠ, Σ²)
 end
 
+
+@inline function calc_nonlinear_κᶜᶜᶜ(i, j, k, grid, closure::SmagorinskyLilly, buoyancy, U, C, ::Val{tracer_index}) where {tracer_index}
+    νₑ = calc_nonlinear_νᶜᶜᶜ(i, j, k, grid, closure, buoyancy, U, C)
+
+    @inbounds Pr = closure.Pr[tracer_index]
+
+    return νₑ / Pr
+end
+
+
```

Comment:
I am not sure this is needed, `κₑ` is given as a `BinaryOperation` so calculating `νₑ` should be enough

Reply:
My thinking here is that it'd be good to unify the interface across models. I was also planning on creating dispatches of this function for `ScalarDiffusivity()` and tuple closures. It'd make things like these easier (in Oceanostics): https://github.com/tomchor/Oceanostics.jl/blob/29a544d3decd833d9f86da05b66a7392197c3b93/src/Oceanostics.jl#L25-L39

For example, as a user, I don't know what function I should use to get the diffusivities and viscosities that works for different closures. As far as I know there isn't one. This is my attempt to get us closer.

Am I missing something?

Reply:
Ok I agree, but I would also leave that `calc_nonlinear_κᶜᶜᶜ` is used to calculate the diffusivity while `κᶜᶜᶜ` is used to retrieve the diffusivity

Reply:
Unity may be a good goal, but we shouldn't add code until the moment we need it / without tests.